modules/c++/nitf/unittests/test_tre_mods++.cpp

```diff
@@ -0,0 +1,129 @@
+/* =========================================================================
+ * This file is part of NITRO
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2019, MDA Information Systems LLC
+ *
+ * NITRO is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <import/nitf.hpp>
+#include <iostream>
+#include "TestCase.h"
+
+namespace
+{
+TEST_CASE(setFields)
+{
+    //create an ACFTA TRE
+    nitf::TRE tre("ACFTA");
+
+    //set a field
+    tre.setField("AC_MSN_ID", "fly-by");
+    TEST_ASSERT_EQ(tre.getField("AC_MSN_ID").toString(),
+                   std::string("fly-by    "));
+
+    //re-set the field
+    tre.setField("AC_MSN_ID", 1.2345678);
+    TEST_ASSERT_EQ(tre.getField("AC_MSN_ID").toString(),
+                   std::string("1.2345678 "));
+
+    //try setting an invalid tag
+    try
+    {
+        tre.setField("invalid-tag", "some data");
+    }
+    catch (const except::Throwable& t)
+    {
+        TEST_ASSERT(false);
+    }
+}
+
+TEST_CASE(cloneTRE)
+{
+    nitf::TRE tre("JITCID");
+    tre.setField("FILCMT", "fyi");
+
+    // try cloning the tre
+    nitf::TRE dolly = tre.clone();
+
+    //the two should NOT be equal -- underlying object is different
+    TEST_ASSERT(tre != dolly);
+    TEST_ASSERT_EQ(tre.getField("FILCMT").toString(),
+                   std::string("fyi"));
+}
+
+TEST_CASE(basicIteration)
+{
+    nitf::TRE tre("ACCPOB");
+
+    {
```

Comment:
Extra set of curly braces on lines 73, 83, 84, 94... there's nothing you're trying to scope here right?  If not, remove.